src/components/widgets/accounts-cash-projection/accounts-cash-projection.directive.coffee

```diff
@@ -65,47 +79,59 @@ module.controller('WidgetAccountsCashProjectionCtrl', ($scope, $q, $filter, $tim
       { expected_payment_date: moment(date).format('YYYY-MM-DD') }
     ).then(-> $scope.trxList.updated = true)
 
-  # Widget specific methods
-  # --------------------------------------
-  w.initContext = ->
-    # TODO: what to do when the widget has no data?
-    $scope.isDataFound = w.content?
-
-    # Offset will be applied to all intervals after today
-    todayInterval = _.findIndex w.content.chart.series[0].data, (dataMat) ->
-      dataMat[0] >= todayUTC
-    $scope.intervalsCount = w.content.chart.series[0].data.length - todayInterval
-
-    projectedSerie = _.find w.content.chart.series, (serie) ->
-      serie.name == "Projected cash"
-
-    cashFlowSerie = _.find w.content.chart.series, (serie) ->
-      serie.name == "Cash flow"
-    cashFlowSerie.data = []
-    cashFlowSerie.type = 'area'
-    cashFlowSerie.showInLegend = false
-
-    totalOffset = 0.0
-    if w.metadata.offset && w.metadata.offset.current && w.metadata.offset.current.length > 0
-      totalOffset += _.sum(w.metadata.offset.current)
-
-    if w.metadata.offset && w.metadata.offset.per_interval && w.metadata.offset.per_interval.length > 0
-      totalOffset += _.sum(w.metadata.offset.per_interval)
+  $scope.trxList.changeResourcesType = (resourcesType) ->
+    return if resourcesType == $scope.trxList.resources
+    $scope.trxList.resources = resourcesType
+    $scope.trxList.fetch()
 
-    if projectedSerie?
-      $scope.currentProjectedCash = projectedSerie.data[todayInterval] - totalOffset
-
-    $scope.isTimePeriodInThePast = w.metadata.hist_parameters && moment(w.metadata.hist_parameters.to) < moment().startOf('day')
-
-    if hist = w.metadata.hist_parameters
-      $scope.fromDate = hist.from
-      $scope.toDate = hist.to
+  $scope.trxList.deleteTransaction = (resourcesType, trxId) ->
+    _.remove($scope.trxList.transactions, (trx) -> trx.id == trxId)
+    BoltResources.destroy(
+      w.metadata.bolt_path,
+      resourcesType,
+      trxId
+    ).then(-> $scope.trxList.updated = true)
 
-  # Timestamps stored in the back-end are in UTC => the filter on the date must be UTC too
-  dateFilter = (timestamp) ->
-    pickedDate = moment.utc(timestamp)
-    if pickedDate <= todayUTC then "lte #{pickedDate.format('YYYY-MM-DD')}" else pickedDate.format('YYYY-MM-DD')
+  # == Sub-Components - Threshold KPI =============================================================
+  $scope.chartDeferred = $q.defer()
+  $scope.chartPromise = $scope.chartDeferred.promise
+  $scope.chartThresholdOptions = {
+    label: 'Get alerted when the cash projection goes below'
+  }
 
+  # == Sub-Components - Add Forecast Popup ========================================================
```

Comment:
Can we move this into a component? Seems like it's pretty decoupled from this controller anyway.